Std/Data/Fin/Basic.lean

```diff
@@ -48,3 +49,22 @@ def natAdd (n) (i : Fin m) : Fin (n + m) := ⟨n + i, Nat.add_lt_add_left i.2 _
 
 /-- `min n m` as an element of `Fin (m + 1)` -/
 def clamp (n m : Nat) : Fin (m + 1) := ⟨min n m, Nat.lt_succ_of_le (Nat.min_le_right ..)⟩
+
+/-- Folds over `Fin n` from the left: `foldl 3 f x = f (f (f x 0) 1) 2` -/
+@[inline] def foldl (n) (f : α → Fin n → α) (init : α) : α :=
```

Comment:
I was going to ask if this warranted a characterising theorem, relating it to folding over the `List` of `Fin`s... but it seems at this point we don't even have a way to obtain the list of `Fin`s!?

Reply:
Yes, I probably wouldn't have made this PR if there was an easy way to get the list of `Fin`. I think `Fin.foldr` is the easiest way to get that list.

If a characterization theorem is essential, I will probably make these `foldlTR`/`foldrTR` and use `csimp`.

Reply:
I tried it and it's kind of a mess to do that characterizing theorem. I have other fishes to fry so I'm hoping we can do without for now :/